### PR TITLE
Implement additional data checks

### PR DESCRIPTION
**Pull request by magik6k**
WIP Changes for https://github.com/filecoin-project/lotus/pull/1124

### CODE REVIEW

File: files.go [hunk 1]
```diff
@@ -81,6 +81,21 @@ func (sb *SectorBuilder) TrimCache(sectorID uint64) error {
...
+		return false, xerrors.Errorf("getting cache dir: %w", err)
+	}
+
+	ents, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return false, err
+	}
+
+	// TODO: slightly more sophisticated check
+	return len(ents) == 10, nil
```
Comment (laser) [inline files.go:96]: What's the significance of the number 10 here? Is this the quantity of artifacts you expect rust-fil-proofs to have written to the cache dir?

Comment (laser) [reply files.go:96]: If so, I think this check should be written in Rust and exposed through CGO. I think it makes sense to put the cache-is-healthy check very close to the place where we write items to the cache. Those two operations (cache health-checking and cache writing) are tightly coupled, and colocating the operations would make that coupling quite clear.

Comment (whyrusleeping) [reply files.go:96]: Yeah, we definitely need a `cacheIsHealthy` call from rust. Can you write that up for us @laser ?

File: sectorbuild_cgo.go [hunk 2]
```diff
@@ -137,6 +138,42 @@ func (sb *SectorBuilder) ReadPieceFromSealedSector(sectorID uint64, offset uint6
 func (sb *SectorBuilder) SealPreCommit(ctx context.Context, sectorID uint64, ticket SealTicket, pieces []PublicPieceInfo) (RawSealPreCommitOutput, error) {
 	fs := sb.filesystem
 
+	cacheDir, err := sb.sectorCacheDir(sectorID)
```
Comment (whyrusleeping) [inline sectorbuild_cgo.go:141]: I would move this checking logic into a separate function, just to keep it clean

Comment (whyrusleeping) [reply sectorbuild_cgo.go:141]: hrm... on second thought, we're using these values later in the function. This is fine for now

**Review by whyrusleeping — APPROVED** [review]
This LGTM, having these check functions be deeper down in rust would be nice, but i wouldnt necessarily block merging on it.

File: files.go (hunk 1 above)
Comment (dignifiedquire) [reply files.go:96]: Yeah, please don't hardcode this number, as the number of files is definitely going to fluctuate and not a stable interface

**Comment by laser** [comment]
Yep, surely. I’ll write something on Monday.

**Comment by magik6k** [comment]
Merging as it's already in lotus master, and makes life annoying to have it be unmerged

**Pull request merged**